src/java/org/apache/cassandra/index/sai/disk/v1/LongArray.java

```diff
@@ -37,6 +37,13 @@ public interface LongArray extends Closeable
      */
     long length();
 
+    /**
+     * Using the given token returns the first row ID corresponding to the token.
+     * @param targetToken Token to lookup and it must not be smaller than previous value
+     * @return The row ID of the given token or negative value if target token is greater than all tokens
+     */
+    long findTokenRowID(long targetToken);
```

Comment:
We've gone back and forth about naming the methods when we have tokens in some cases and not tokens in others, so I don't think you have to change anything at this point, especially if we're going to make further changes to potentially eliminate the sorted terms entirely. We name this method `indexOf(long value)` or something like that, kind of following the `get()`/`indexOf()` pattern you typically see in Java collections. (see `List#indexOf()`)

The callers in the `rowIdFromPrimaryKey()` methods make the usage pretty clear, as does the thing they access, usually called `tokenArray`.

Reply:
It also would make `PostingsReader.LongArrayReader` less odd I guess.

Reply:
That naming does make more sense. It feels in place.